addon/index.ts

```diff
@@ -22,45 +25,60 @@ window.addEventListener('storage', function ({ key, newValue }) {
     return;
   }
 
+  // skip if newValue is null
+  if (newValue === null) {
+    return;
+  }
+
   localStorageCache.set(key, jsonParseAndFreeze(newValue));
 });
 
-export default function localStorageDecoratorFactory(...args) {
+export default function localStorageDecoratorFactory(
+  ...args: DecoratorArgs
+): void;
```

Comment:
The direct invocation overload needs `void` return type or this error is seen:

`Decorator function return type is 'PropertyDescriptor' but is expected to be 'void' or 'any'`